ncwms_configurator/ncwms_configurator.py

```diff
@@ -0,0 +1,219 @@
+import sys
+import os
+import logging
+from argparse import ArgumentParser
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from modelmeta import DataFile, DataFileVariable, Ensemble, EnsembleDataFileVariables
+
+from lxml import etree
+
+log = logging.getLogger(__name__)
+
+def get_variable(id, title, colorScaleRange, palette="rainbow", scaling="linear", numColorBands="250", disabled="false"):
+    return etree.Element(
+        "variable",
+        id = id,
+        title = title,
+        colorScaleRange = colorScaleRange,
+        palette = palette,
+        scaling = scaling,
+        numColorBands = numColorBands,
+        disabled = disabled
+    )
+
+
+def get_dataset(id, location, title, queryable="true", dataReaderClass="", copyrightStatement="", moreInfo="", disabled="false", updateInterval="-1"):
+    return etree.Element(
+        "dataset",
+        id = id,
+        location = location,
+        title = title,
+        queryable = queryable,
+        dataReaderClass = dataReaderClass,
+        copyrightStatement = copyrightStatement,
+        moreInfo = moreInfo,
+        disabled = disabled,
+        updateInterval = updateInterval
+    )
+
+
+
+
+def get_contact(name = "", organization = "", telephone = "", email = ""):
+    root = etree.Element("contact")
+    etree.SubElement(root, "name").text = name
+    etree.SubElement(root, "organization").text = organization
+    etree.SubElement(root, "telephone").text = telephone
+    etree.SubElement(root, "email").text = email
+    return root
+
```

Comment:
Yeah, it could be. Part of the reason I was more explicit with these (I originally had Config take kwargs but didn't follow through with it) is that I wanted to enforce the element names and the default arguments. The element names can be easily checked, but I couldn't think of a 'good' way at the time to set default arguments when using kwargs with different sets of required/optional/default arguments.

Thinking about it more: For your first suggestion, we could just use dict.update and have a default set specific to that element. For the second simplification, we'd have to conditionally set the default dict, check 'required' kwargs, and generate the element as appropriate.

I think I've talked my way into thinking it would be shorter/more maintainable :+1: